### DISCUSSION TITLE
Point is located in two suburbs even when it isn't

### DISCUSSION THREAD

**Discussion by Dimitar5555** [osm-search/Nominatim#3284] [Q&A] [answered]
## What did you search for?

Show address at 42.67199 23.30508 and zoom level 16 or higher https://www.openstreetmap.org/search?whereami=1&query=42.67199%2C23.30508#map=18/42.67190/23.30525

## What result did you get?

[ж.г. Южен парк](https://www.openstreetmap.org/node/9590318772) and [ж.к. Иван Вазов](https://www.openstreetmap.org/relation/14580256)

## What result did you expect?

**When the result in the right place and just named wrongly:** 

**When the result missing completely:**

bul. Petko Todorov (https://www.openstreetmap.org/way/259443696) or the park itself (https://www.openstreetmap.org/relation/12739635).

## Further details

In this area all boundaries of suburbs are mapped as relations. I know that Nominatim doesn't like broken polygons but both look good. The park also has a boundary (https://www.openstreetmap.org/relation/16878152) but it seems to be ignored by Nominatim.

Running the query on zoom level 15 ([link](https://www.openstreetmap.org/search?whereami=1&query=42.67199%2C23.30508#map=15/42.6720/23.3051)) results in a completely diffrent result but it's still not correct.

**Comment by mtmail** [comment, marked as answer]
I can explain Nominatim's logic a bit.

Nominatim will look for the nearest named feature and return the coordinates and data for that.

https://nominatim.openstreetmap.org/ui/reverse.html?lat=42.67190&lon=23.30525&zoom=18

The red circle is the search position, the blue circle is the result.

![image](https://github.com/osm-search/Nominatim/assets/3727288/b7fdebe4-29b2-45f4-98fd-255fc3fe84e4)

The result is a relation of many paths. https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=14925862&class=highway Now we need to look at the 'address table'.

![image](https://github.com/osm-search/Nominatim/assets/3727288/0033e9ba-21b4-44ae-93b9-493526bd079a)

The leisure=park (https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=12739635) is not in the address list. Parks are not addresses unless the result found feature is a park. If you strictly need "anything that covers these coordinates" then the Overpass API is a better tool.

As you see it finds a place=quarter and a place=suburb for the address. If they were the same place type (same address rank) then it would select only one.

For both it lists the node only. That means Nominatim doesn't know the exact boundaries (yet). It looks at distance to the node and makes a guess that the found position belong to the node.

For place=quarter Yuzhen park (ж.г. Южен парк) Nominatim hasn't linked node https://www.openstreetmap.org/node/3918348126 and relation https://www.openstreetmap.org/relation/16868071 together. Try setting the same wikidata tag on both and Nominatim should treat both as the same place.

For place=suburb Ivan Vazov (ж.к. Иван Вазов) I see the wikidata is already set on the node https://www.openstreetmap.org/node/1699545833 and relation https://www.openstreetmap.org/relation/14580256 It should have linked them already. It's possible the area needs to be reprocessed or reindexed.

I would set `admin_level` tags on the suburbs and quarters (the relations, not the nodes), I can't see any set to > 8.

If Nominatim finds the quarter and suburb boundaries then the result will be better, both will be excluded from the address list of the paths. I don't think it will link the paths to the park though. Is the park part of any suburb or quarter?

**Reply by Dimitar5555** [reply]
> I would set admin_level tags on the suburbs and quarters (the relations, not the nodes), I can't see any set to > 8.

Thanks for the idea, that seems to have fixed the issue. Now it says that's in Triaditsa, which covers the park's area.

**Comment by Dimitar5555** [comment]
Thanks for the fast and detailed reply!
> Nominatim will look for the nearest named feature and return the coordinates and data for that.
>
> https://nominatim.openstreetmap.org/ui/reverse.html?lat=42.67190&lon=23.30525&zoom=18
>
>The result is a relation of many paths. https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=14925862&class=highway Now we need to look at the 'address table'.

That relation is not named. Why would Nominatim choose it over the nearby named road ([bul. Petko Todorov](https://www.openstreetmap.org/way/259443696))?

> For place=quarter Yuzhen park (ж.г. Южен парк) Nominatim hasn't linked node https://www.openstreetmap.org/node/3918348126 and relation https://www.openstreetmap.org/relation/16868071 together. Try setting the same wikidata tag on both and Nominatim should treat both as the same place.

That's new info to me. Usually it seems to link them based on the relation without having wikidata tags.

I've added wikidata tags to Yuzhen park area and admin_level=9 to both areas (8 is used for city districts and town borders in Bulgaria).

> ... I don't think it will link the paths to the park though. Is the park part of any suburb or quarter?

The park is not part of other suburbs or quarters, but if you are there and somebody asks you where are you, you will tell them that you are in that park, not in the nearby residential complex.

**Reply by mtmail** [reply]
> That [relation](https://www.openstreetmap.org/relation/14925862) is not named.

It's treated like an unnamed street. A market place or pedestrian zone in a city are similar.

> The park is not part of other suburbs or quarters, but if you are there and somebody asks you where are you, you will tell them that you are in that park, not in the nearby residential complex.

That's understandable but Nominatim, at least the installation on nominatim.openstreetmap.org, isn't a good choice for 'where am i'. The Overpass API (the '?' feature on openstreetmap.org) is better at finding all enclosing features.

![image](https://github.com/osm-search/Nominatim/assets/3727288/4500f748-50a4-446a-9722-ad3b65708818)

I'd make the park a administrative area, maybe it's like a neighborhood?.

BTW did you notice one suburb seems to exist twice?

1. https://www.openstreetmap.org/relation/16867359
ж.к. Манастирски ливади
2. https://www.openstreetmap.org/relation/16859317
ж.к. Манастирски ливади - изток

**Reply by Dimitar5555** [reply]
> BTW did you notice one suburb seems to exist twice?

It exists five times (or there are five areas with `name~Манастирски ливади`). It's a long story but I think that it's fixed now.
